R/summary_funs.R

```diff
@@ -202,198 +209,223 @@ weighted_summary_means <- function(y_wobs_test, family, wdraws, mu, dis, cl_ref,
+      # simple transformation of mse
+      value <- sqrt(mse_e)
+      # the first-order Taylor approximation of the variance
+      value_se <- sqrt(value_se^2 / mse_e / 4)
+    } else if (stat == "R2") {
+      # simple transformation of mse
+      mse_y <- mean(wcv * (mean(y)-y)^2)
+      value <- 1 - mse_e / mse_y
+      # the first-order Taylor approximation of the variance
+      var_mse_y <- .weighted_sd((mean(y)-y)^2, wcv)^2 / n
+      if (is.null(summaries_fast) || sum(n_loo==n)) {
+        if (is.null(summaries_baseline)) {
+          cov_mse_e_y <- mean(wcv * ((mu - y)^2 - mse_e) *
+                                ((mean(y)-y)^2-mse_y)) / n
+        } else {
+          cov_mse_e_y <- mean(wcv * ((mu - y)^2 - mse_e -
+                                       ((mu_baseline - y)^2 - mse_b)) *
+                                ((mean(y)-y)^2-mse_y)) / n
+        }
       } else {
-        value <- sqrt(mean(wcv * (mu - y)^2, na.rm = TRUE))
-        value.bootstrap <- bootstrap(
-          (mu - y)^2,
-          function(resid2) {
-            sqrt(mean(wcv * resid2, na.rm = TRUE))
-          },
-          ...
-        )
-        value.se <- sd(value.bootstrap)
-        lq_uq <- quantile(value.bootstrap,
-                          probs = c(alpha_half, one_minus_alpha_half),
-                          names = FALSE, na.rm = TRUE)
+        mse_e_fast <- mean(wcv * (summaries_fast$mu-y)^2)
+        if (is.null(summaries_baseline)) {
+          srs_diffe <-
+            .srs_diff_est_w(y_approx =
+                              ((summaries_fast$mu - y)^2 - mse_e_fast) *
+                              ((mean(y)-y)^2 - mse_y),
+                            y = (((mu - y)^2 - mse_e) *
+                                   ((mean(y)-y)^2 -mse_y))[loo_inds],
+                            y_idx = loo_inds,
+                            w = wcv)
+        } else {
+          srs_diffe <-
+            .srs_diff_est_w(y_approx =
+                              ((summaries_fast$mu - y)^2 - mse_e_fast-
+                                 ((mu_baseline - y)^2 - mse_b)) *
+                              ((mean(y)-y)^2 - mse_y),
+                            y = (((mu - y)^2 - mse_e -
+                                    ((mu_baseline - y)^2 - mse_b)) *
+                                   ((mean(y)-y)^2 -mse_y))[loo_inds],
+                            y_idx = loo_inds,
+                            w = wcv)
+        }
+        cov_mse_e_y <- srs_diffe$y_hat / n^2
+      }
+      # part of delta se comes automatically via mse
+      var_mse_e <- value_se^2
+      if (!is.null(summaries_baseline)) {
+        # delta=TRUE
+        mse_e <- mse_e - mse_b
       }
+      value_se <- sqrt((var_mse_e -
+                          2 * mse_e / mse_y * cov_mse_e_y +
+                          (mse_e / mse_y)^2 * var_mse_y) / mse_y^2)
     }
   } else if (stat %in% c("acc", "pctcorr", "auc")) {
     y <- y_wobs_test$y
+    wcv <- y_wobs_test$wobs
+    wcv <- n * wcv / sum(wcv)
```

Comment:
Shouldn't these two lines be replaced by `wcv <- rep(1, n)` here?

Reply:
Why?

Reply:
In the `if (stat %in% c("acc", "pctcorr", "auc"))` case, `y_wobs_test$wobs` can only be an integer vector. Furthermore, in the `if (!is.null(y_wobs_test$y_prop))` sub-case, `y_wobs_test$wobs` can indeed contain positive integers. In that case, the aggregated Bernoulli trials are de-aggregated, producing a longer dataset, and we define `wcv <- rep(wcv, y_wobs_test$wobs)`. Thus, `wcv` needs to be all-`1`s there. In the `else` sub-case (i.e., "`if (is.null(y_wobs_test$y_prop))`"), `y_wobs_test$wobs` can only contain `1`s and hence `wcv` needs to be all-`1`s there as well.

Reply:
Thanks for the explanation

Reply:
replaced